### ISSUE TITLE
Issue with cached images

### ISSUE THREAD

**Issue by luisgserrano** [nikkomiu/phoenix_inline_svg#19]
Hi,

I've tried to use your library but I'm getting some issues.

I've changed the default dir for images:
- `config :phoenix_inline_svg, dir: "/priv/static/images"`

I'm using the 'new' version:
- `use PhoenixInlineSvg.Helpers, otp_app: :my_app`

But when I try to build the app, it gives me this error:
```
(ArgumentError) argument error
    :erlang.hd(nil)
    lib/phoenix_inline_svg/helpers.ex:227: PhoenixInlineSvg.Helpers.create_cached_svg_image/1
    (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2
    (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2
    expanding macro: PhoenixInlineSvg.Helpers.__using__/1
```

I'm importing like this:
`<%= svg_image("image_name", "some_inner_folder", class: "some-class") %>`

I already tried adding that file to cache svgs but didn't work.
Do you know what I'm doing wrong?

**Comment by mhanberg** [comment]
Not at the computer rn, but I think this is a bug. 

But to clarify, what directory is the “image_name.svg” located in?

**Comment by luisgserrano** [comment]
In `some_inner_folder`, inside `/priv/static/images`.

**Comment by nikkomiu** [comment]
A little more clarification: did you make sure to change the `my_app` part in the `use PhoenixInlineSvg.Helpers, otp_app: :my_app` to the name of your app? -- Nevermind there is a different error for that sorry

**Comment by mhanberg** [comment]
For full clarity, it is in `/priv/static/images/some_inner_folder/image_name.svg`?

**Comment by luisgserrano** [comment]
yes @mhanberg !

**Comment by luisgserrano** [comment]
@nikkomiu yes I've changed to the name of my app

**Comment by mhanberg** [comment]
@luisgserrano I was not able to reproduce this.

Could you take a look at this example repository and confirm that your code matches it?

https://github.com/mhanberg/inline_svg_test/commit/1c472a53681d8033e6d75707acd82e4238dab6a1

**Comment by luisgserrano** [comment]
@mhanberg I did like your test again and it gives me the same error as I've posted in the first comment. The app doesn't build when calling `mix phx.server`

**Comment by mhanberg** [comment]
I wasn't able to reproduce the same stack trace, but i think i figured it out.

Webpack (or whatever JS tool you are using) is responsible for copying your files from `assets/static` to `priv/static`.

Webpack will do this _after_ mix compiles your application, so if they aren't copied over to priv, then the library can't find them.

I would consider this a bug.

As a work around, you can `mv assets/static/images priv/static/images` and then attempt to `mix phx.server` in development.

**Comment by nikkomiu** [comment]
I’m able to reproduce this. I may have the reason too but I’ll have to try it later today or tomorrow to validate it.

**Comment by mhanberg** [comment]
I think the solution is to pull the images from `assets` instead of `priv`.

**Comment by nikkomiu** [comment]
The issue I ran into is when there is a file that is in the directory that does not match the `Regex.run` and the `hd` (at `lib/phoenix_inline_svg/helpers.ex:227`) command throws an exception because the list is empty.

I have the fix locally. I will create a branch and push it later today with the fix.

**Comment by nikkomiu** [comment]
I also noticed that after the dir tree is traversed the`create_cached_svg_image` fn is called multiple times per image. I'm going to try to fix that too before releasing a fix for this issue.

**Comment by luisgserrano** [comment]
Cool man! Thanks @nikkomiu

**Comment by nikkomiu** [comment]
@luisgserrano can you try this branch `bugfix/catch-file-error` when you get a chance and see if it fixes your issue?

**Comment by psb** [comment]
Hi,

I have the same problem and I tried to use your bugfix branch; however, I'm new to Elixir and Phoenix so I don't if I used it properly. I did the following and it kind of works:

1. Deleted the _build folder of my project;
2. Rebuilt my project;
3. Compiled your bugfix branch;
4. Copied the bugfix ebin folder to the phoenix_inline_svg folder in my project. (I only copied the ebin folder because copying the .mix folder as well resulted in the same original issue.);
5. Added excoveralls and inch_ex to my projects deps because phx.server complained about them not being present;
6. Rebuilt my project using phx.server.

Loading the svgs "the new way" (i.e. with caching) now works. (vscode with elixir_ls still complains about the original issue with an error in a random view file, but phx.server is fine.)

Using the plugin "the old way" (i.e. without caching) always worked.

**Comment by mhanberg** [comment]
@psb 

The method I recommend for trying out the branch is replacing the version in your mix.exs with the github and branch options.

`{:phoenix_inline_svg, github: "mhanberg/phoenix_inline_svg", branch: "fix-generated-functions"},`

After that, you should be able to run `mix deps.get` and `mix compile`.

**Comment by psb** [comment]
@mhanberg Thanks for the tip. I knew you could use github in your deps, but I didn't know about the branch option.

Using `{:phoenix_inline_svg, github: "nikkomiu/phoenix_inline_svg", branch: "bugfix/catch-file-error"}` in my deps and rebuilding makes the plugin work perfectly. Even vscode and elixir_ls have stopped complaining 😄

**Comment by mhanberg** [comment]
@psb I'm glad you got it working!

Would you mind trying something out for me?

Can you delete the `priv/static` directory and then recompile? I think there may still be a problem that #20 solves.

Thanks!

**Comment by psb** [comment]
I deleted the `priv/static` folder and recompiled; everything seems to be working fine.

**Comment by nikkomiu** [comment]
Fixed on v1.3.1. Please reopen this issue if there are still problems after updating.
